tmol/score/cartbonded/potentials/potentials.hh

```diff
@@ -38,6 +38,56 @@ def cblength_V_dV(Real3 atm1, Real3 atm2, Real K, Real x0)
   return {E, dE * dist.dV_dA, dE * dist.dV_dB};
 }
 
+template <typename Real>
+def cblength_V_dV2(Real3 atm1, Real3 atm2, Real K, Real x0)
+    ->tuple<Real, Vec<Real3, 2>> {
+  auto dist = distance<Real>::V_dV(atm1, atm2);
+  Real E = 0.5 * K * square(dist.V - x0);
+  Real dE = K * (dist.V - x0);
+  return {E, {dE * dist.dV_dA, dE * dist.dV_dB}};
```

Comment:
This is what is causing CUDA grief. This compiles and runs just fine on CPU. CUDA complains:

```
E   /home/jflat06/rosetta/tmol/tmol/score/cartbonded/potentials/potentials.hh(49): error: no instance of constructor "mgpu::tuple<args_t...>::tuple [with args_t=<float, Eigen::Matrix<Eigen::Matrix<float, 3, 1, 0, 3, 1>, 2, 1, 0, 2, 1>>]" matches the argument list
E               argument types are: (float, {...})
E             detected during:
E               instantiation of "auto tmol::score::cartbonded::potentials::cblength_V_dV2(tmol::score::cartbonded::potentials::Vec<Real, 3>, tmol::score::cartbonded::potentials::Vec<Real, 3>, Real, Real)->mgpu::tuple<Real, tmol::score::cartbonded::potentials::Vec<tmol::score::cartbonded::potentials::Vec<Real, 3>, 2>> [with Real=float]" 
```

Does CUDA not like the nested Eigen::Matrices? Or does it just fail to handle the nested {} notation?

Reply:
Pretty sure it's the nested {}.  You need to initialize return val as Vec<Real3, 2> and use .col() to set each column separately.  e.g., in kinematic code:
>     ht.col(3) = Eigen::Matrix<Real, 1, 4>(0, 0, 0, 1);

Reply:
Yeah that was what I was avoiding, as I think it makes things less readable. I'll give it a try.